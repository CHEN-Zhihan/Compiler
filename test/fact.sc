def fact(n) {
  if(n<1) return 1;
  return n * fact(n-1);
}
puts_("Please enter a +ve int <= 12: ");
n = geti();
if (n <= 12) {
	puti(fact(n));
} else {
	puti_(n); puts(" > 12!!");
}
