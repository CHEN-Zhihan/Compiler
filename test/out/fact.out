Please enter a +ve int <= 12: 120

---
Please enter a +ve int <= 12: 2

---
Please enter a +ve int <= 12: 720
